### ISSUE TITLE
sentences() hangs the process with some examples

### ISSUE THREAD

**Issue by sosloow** [nlp-compromise/compromise#380]
Hi, we tried to switch to the new version of the library in our project, and got this problem from the get-go. In a test running on a small sample some sentences caused process to hang. As we later discovered, sentences() was the cause of this problem.
Here's an example freezing the page: https://jsfiddle.net/hqq14q39/

And by the way, can I ask you, we used the library mostly for bringing everything in the text into root forms (like this: `nlp.text(phrase).contractions.expand().root()`) and then splitting them into terms. What can be a proper way to do this with the new API? Thanks.

**Comment by spencermountain** [comment]
yikes!
thanks Stepan, will check it out this afternoon.

yeah, this is pretty doable in the new version, too
```
var nlp=require('compromise')
var doc=nlp(phrase)
doc.contractions.expand()
doc.nouns().toSingular()
doc.verbs().toInfinitive()
//.....
console.log(doc.out('normal'))
```

**Comment by spencermountain** [comment]
hey, this was a nice one. thanks
(fixed in `10.5.1`)
